doc/reference/reference_rock/vshard/vshard_ref.rst

```diff
@@ -491,7 +519,74 @@ Router public API
 
     .. code-block:: lua
 
-        bucket_id = vshard.router.bucket_id(18374927634039)
+        bucket_id = vshard.router.bucket_id_strcrc32(18374927634039)
+
+    .. Note::
+
+        Remember that it is not safe. See details in :ref:`bucket_id() <router_api-bucket_id>`
+
+.. _router_api-bucket_id_mpcrc32:
+
+.. function:: vshard.router.bucket_id_mpcrc32(key)
+
+    This function is safer than ``bucket_id_strcrc32``. It takes a CRC32 from
+    MessagePack encoded value. That is, bucket id of integers does not
+    depend on their Lua type. In case of a string key does not encode it into
+    MessagePack, but takes hash right from the string. 
+
+    :param key: a hash key. This can be any Lua object (number, table, string).
+
+    :Return: a bucket identifier
+    :Rtype: number
+
+    However it still may return different values for not equal floating point
+    types. That is, ``ffi.cast('float', number)`` may be reflected into a bucket id
+    not equal to ``ffi.cast('double', number)``. This can't be fixed, because a
+    float value, even being casted to double, may have a garbage tail in its fraction.
+
+    Floating point keys should not be used to calculate a bucket id,
+    usually.
+
+    Be very careful in case you store floating point types in a space. When data
+    is returned from a space, it is cased to Lua number. And if that value had
```

Comment:
cased? please double-check, must be "cast" I suppose (past participle)

Reply:
```suggestion
    is returned from a space, it is cast to Lua number. And if that value had
```